engine/runtime/src/main/java/org/enso/interpreter/runtime/EnsoContext.java

```diff
@@ -135,7 +135,9 @@ public EnsoContext(
     this.isIrCachingDisabled =
         getOption(RuntimeOptions.DISABLE_IR_CACHES_KEY) || isParallelismEnabled;
     this.executionEnvironment = getOption(EnsoLanguage.EXECUTION_ENVIRONMENT);
-
+    if (getOption(RuntimeOptions.ENABLE_ASSERTIONS_KEY)) {
+      this.assertsDisabledAssumption.invalidate("Enso assertions are enabled");
```

Comment:
Done this way, you don't need an `Assumption`, you'd be fine with `final boolean`.

`Assumption` is only good if its assumption can change during runtime. Do we have a way, example or test that changes the _"assertion assumption"_ during runtime?

Reply:
Good point. Changing "assertions enabled" during runtime does not make sense. Reimplemented this via `final boolean` field instead of `Assumption` in b435b75360098870a8973ac600c81472ec623d2e.